File: app/src/main/java/com/littlefox/app/foxschool/object/result/MainInformationResult.java
package com.littlefox.app.foxschool.object.result;

import net.littlefox.lf_app_fragment.object.result.common.ContentsBaseResult;
import net.littlefox.lf_app_fragment.object.result.common.SeriesBaseResult;
import net.littlefox.lf_app_fragment.object.result.littlefoxClass.ClassMainResult;

import java.util.ArrayList;

public class MainInformationResult
{
    private MainStoryInformationResult story = null;
    private MainSongInformationResult song = null;
    private ArrayList<MyBookshelfResult> bookshelves = new ArrayList<MyBookshelfResult>();
    private ArrayList<MyVocabularyResult> vocabularies = new ArrayList<MyVocabularyResult>();
    private CompanyInformationResult company_information = null;
    private InAppCompaignResult in_app_campaign = null;

    public MainStoryInformationResult getMainStoryInformation()
    {
        return story;
    }

    public MainSongInformationResult getMainSongInformation()
    {
        return song;
    }

    public ArrayList<MyBookshelfResult> getBookShelvesList()
    {
        return bookshelves;
    }

    public ArrayList<MyVocabularyResult> getVocabulariesList()
    {
        return vocabularies;
    }

    public CompanyInformationResult getCompanyInformation()
    {
        return company_information;
    }

    public InAppCompaignResult getInAppCompaignInformation()
    {
        return in_app_campaign;
    }

}

File: app/src/main/java/com/littlefox/app/foxschool/object/result/main/MainInformationResult.kt
package com.littlefox.app.foxschool.`object`.result.main

import com.google.gson.annotations.SerializedName
import com.littlefox.app.foxschool.`object`.result.main.*
import com.littlefox.app.foxschool.`object`.result.story.SeriesInformationResult
import java.util.*

class MainInformationResult
{
    @SerializedName("story")
    private val story : MainStoryInformationResult? = null

    @SerializedName("song")
    private val song : ArrayList<SeriesInformationResult> = ArrayList<SeriesInformationResult>()

    @SerializedName("bookshelves")
    private val bookshelves : ArrayList<MyBookshelfResult> = ArrayList<MyBookshelfResult>()

    @SerializedName("vocabularies")
    private val vocabularies : ArrayList<MyVocabularyResult> = ArrayList<MyVocabularyResult>()

    @SerializedName("company_information")
    private val company_information : CompanyInformationResult? = null

    @SerializedName("in_app_campaign")
    private val in_app_campaign : InAppCompaignResult? = null

    @SerializedName("homework")
    private val homework : String  = ""

    @SerializedName("news")
    private val news : String      = ""

    @SerializedName("files")
    private val files : FileInformationResult? = null

    fun getMainStoryInformation() : MainStoryInformationResult = story!!

    fun getMainSongInformationList() : ArrayList<SeriesInformationResult> = song

    fun getBookShelvesList() : ArrayList<MyBookshelfResult> = bookshelves

    fun getVocabulariesList() : ArrayList<MyVocabularyResult> = vocabularies

    fun getCompanyInformation() : CompanyInformationResult? = company_information

    fun getInAppCompaignInformation() : InAppCompaignResult? = in_app_campaign

    val isUpdateHomework : Boolean
        get()
        {
            if(homework.equals("Y"))
            {
                return true
            }
            return false
        }

    val isUpdateNews : Boolean
        get()
        {
            if(news.equals("Y"))
            {
                return true
            }
            return false
        }

    fun getFileInformation() : FileInformationResult? = files
}